modules/kubernetes/external-dns/variables.tf

```diff
@@ -38,3 +38,15 @@ variable "azure_resource_id" {
   type        = string
   default     = ""
 }
+
+variable "aws_region" {
+  description = "AWS region to use"
+  type        = string
+  default     = ""
+}
+
+variable "aws_role_arn" {
+  description = "AWS role ARN to attach to service account"
+  type        = string
+  default     = ""
+}
```

Comment:
I think it would be smart of us to have cloud-provider specific configuration in object, like:
```hcl
variable "aws_config" {
  description = "AWS role ARN to attach to service account"
  type        = object({aws_role_arn=string, aws_region=string})
  default     = {aws_role_arn="", aws_region=""}
}
```

Then we can have one object per provider and then it's always configured based on the cloud_provider string which one is used.

Thoughts?

Reply:
done